File: services/scrapper/src/scrapper/datatypes/get_last_chapter.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class GetLastChapterByIdRequest:
    novel_id: int


@dataclass
class GetLastChapterByNameRequest:
    name: str


@dataclass
class GetLastChapterResponse:
    success: bool
    message: Optional[str] = None
    novel_id: Optional[int] = None
    last_chapter_number: Optional[int] = None
    last_chapter_name: Optional[str] = None

File: internal/http/payloads/get_last_chapter.go
package payloads

type GetLastChapterByIdRequest struct {
	NovelID int64 `json:"novel_id"`
}

type GetLastChapterByNameRequest struct {
	Name string `json:"name"`
}

type GetLastChapterResponse struct {
	Success           bool   `json:"success"`
	Message           string `json:"message,omitempty"`
	NovelID           int64  `json:"novel_id,omitempty"`
	LastChapterNumber int64  `json:"last_chapter_number,omitempty"`
	LastChapterName   string `json:"last_chapter_name,omitempty"`
}
